be/src/storage/persistent_index.cpp

```diff
@@ -1063,9 +1063,7 @@ class SliceMutableIndex : public MutableIndex {
     }
 
     // return the dump file size if dump _set into a new file
-    // If _set is empty, the real snapshot file will only wite a size_(type is size_t) into file. So we
-    // will use `sizeof(size_t)` as return value.
-    size_t dump_bound() { return sizeof(size_t) + _total_kv_pairs_usage; }
+    size_t dump_bound() { return sizeof(size_t) + _total_kv_pairs_usage + sizeof(size_t) * size(); }
```

Comment:
```suggestion
    size_t dump_bound() { return (sizeof(size_t) + 1) * size() + _total_kv_pairs_usage; }
```

Reply:
> 

done, update in https://github.com/StarRocks/starrocks/pull/10772/commits/c20ce64b14bcf7cf3065ced2c80bb04e6b35f853